index.ts

```diff
@@ -0,0 +1,572 @@
+  infiltrates?: boolean;
+  isFutureMove?: boolean;
+  isSpread?: boolean | 'AllAdjacent';
+  isZ?: ItemT;
+  multiaccuracy?: boolean;
+  multihit?: [number, number]; // range
+  noCopy?: boolean;
+  noDamageVariance?: boolean;
+  noFaint?: boolean;
+  noMetronome?: MoveT[];
+  noPPBoosts?: boolean;
+  noSketch?: boolean;
+  ohko?: true | 'Ice'; // TODO: better type here?
+  percentHealed?: number;
+  pressureTarget?: string;
+  priority?: number;
+  pseudoWeather?: string:
+  recoil?: number | 'Custom'; // fraction
+  secondaries?: SecondaryEffect[];
+  self?: SelfEffect;
+  selfBoost?: {boosts?: Partial<BoostsTable>};
+  selfSwitch?: ID | boolean;
+  selfdestruct?: 'Always' | 'IfHit';
+  sideCondition?: ID;
+  sleepUsable?: boolean;
+  slotCondition?: string;
+  stallingMove?: boolean;
+  status?: StatusName;
+  stealsBoosts?: boolean;
+  terrain?: string;
+  thawsTarget?: boolean;
+  useSourceDefensive?: boolean;
+  useTargetOffensive?: boolean;
+  usesHighestAttackStat?: boolean;
+  volatileStatus?: ID;
+  weather?: WeatherName;
+  zMove?: {
+    boosts?: Partial<BoostsTable>;
+    effect?: ID;
+    power?: number;
+  }
+}> {}
+
+interface JSONMove extends Nullable<Data & CommonMove<ID, ID, ID>> {}
+interface Move extends DeepReadonly<Data & CommonMove<Item, Move, Type> & {
+  kind: 'Move';
+
+  noMetronome: Move[];
+  secondaries: SecondaryEffect[];
+
+  toJSON(): JSONMove;
+}> {}
+
+interface Moves {
+  get(id: ID): Move | undefined;
+  [Symbol.iterator](): IterableIterator<Move>;
+}
+
+// ---------- species.ts ----------
+
+type GenderName = 'M' | 'F' | 'N';
+type UsageTierName = 'OU' | 'UU' | 'RU' | 'NU' | 'PU';
+type TierName = UsageTierName | 'Uber' | 'UUBL' | 'RUBL' | 'NUBL' | 'PUBL';
+
+interface SpeciesAbilities<AbilityT> extends DeepReadonly<{
+  0: AbilityT;
+  1?: AbilityT;
+  H?: AbilityT;
+  S?: AbilityT;
+}> {};
+
+interface CommonSpecies<AbilityT, ItemT, MoveT, SpeciesT, TypeT> {
+  num: number;
+  abilities: SpeciesAbilities<AbilityT>;
+  baseForme?: SpeciesT;
+  baseSpecies?: SpeciesT;
+  baseStats: StatsTable;
+  battleOnly?: boolean;
+  cosmeticForms?: SpeciesT[];
+  doublesTier?: TierName;
+  eventOnly?: boolean;
+  evos?: SpeciesT[];
+  formeLetter?: string;
+  forme?: string;
+  isMega?: boolean;
+  isPrimal?: boolean;
+  isTotem?: boolean;
+  maleOnlyHidden?: boolean
+  maxHP?: number;
+  nfe: boolean;
+  otherFormes?: SpeciesT[];
+  prevo?: SpeciesT;
+  requiredAbility?: AbilityT;
+  requiredItems?: ItemT[];
+  requiredMove?: MoveT;
+  tier?: TierName;
+  types: TypeT[];
+  unreleasedHidden?: boolean
+  weightkg: number;
+}
+
+interface JSONSpecies extends Nullable<JSONData & CommonSpecies<ID, ID, ID, ID, ID>> {}
+interface Species extends DeepReadonly<Data & CommonSpecies<Ability, Item, Move, Species, Type> & {
+  type 'Species';
+
+  cosmeticForms: Species[];
+  otherFormes: Speces[];
+  requiredItems: Item[];
+
+  toJSON(): JSONSpecies;
+}> {}
+
+interface SpeciesSummaries {
+  getName(name: string): string;
```

Comment:
i don't understand what this is

Reply:
https://github.com/Zarel/Pokemon-Showdown/blob/master/sim/dex.ts#L320-L332 possibly can be dropped